library/core/src/iter/traits/iterator.rs

```diff
@@ -2231,6 +2231,179 @@ pub trait Iterator {
+    /// and so on.
+    ///
+    /// `find_or_nth()` is short-circuiting; in other words, it will stop processing as soon as the
+    /// closure returns `true`.
+    ///
+    /// Because `find_or_nth()` takes a reference, and many iterators iterate over references, this
+    /// leads to a possibly confusing situation where the argument is a double reference. You can
+    /// see this effect in the examples below, with `&&x`.
+    ///
+    /// [`Some(element)`]: Some
+    /// [`nth(n)`]: Iterator::nth
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// #![feature(iter_find_or_fnl)]
+    ///
+    /// let a = [0, 1, 2, 3];
+    ///
+    /// assert_eq!(
+    ///     a.iter().find_or_nth(|&&x| x >  1, 2),
+    ///     Some(&2) // predicate is satisfied
+    /// );
+    /// assert_eq!(
+    ///     a.iter().find_or_nth(|&&x| x > 10, 2),
+    ///     Some(&2) // predicate not satisfied, 3rd element is returned
+    /// );
+    /// assert_eq!(
+    ///     a.iter().find_or_nth(|&&x| x > 10, 5),
+    ///     None // predicate not satisfied, no 6th element
+    /// );
+    /// ```
+    #[inline]
+    #[unstable(feature = "iter_find_or_fnl", reason = "recently added", issue = "none")]
+    fn find_or_nth<P>(&mut self, predicate: P, n: usize) -> Option<Self::Item>
+    where
+        Self: Sized,
+        P: FnMut(&Self::Item) -> bool,
+    {
+        #[inline]
+        fn check<T>(
+            mut predicate: impl FnMut(&T) -> bool,
+            n: usize,
+        ) -> impl FnMut(Option<T>, (usize, T)) -> ControlFlow<T, Option<T>> {
+            move |prev, (idx, cur)| {
+                if predicate(&cur) {
+                    ControlFlow::Break(cur)
+                } else if idx == n {
+                    debug_assert!(prev.is_none());
+                    ControlFlow::Continue(Some(cur))
+                } else {
+                    debug_assert_eq!(idx < n, prev.is_none());
+                    ControlFlow::Continue(prev)
+                }
+            }
+        }
+
+        self.enumerate().try_fold(None, check(predicate, n)).into_result().unwrap_or_else(Some)
+    }
+
+    /// Searches for an element of an iterator that satisfies a predicate, returns the last
+    /// element if no such element is found.
+    ///
+    /// `find_or_last()` takes a closure that returns `true` or `false`. It applies this closure to
+    /// each element of the iterator, and if any of them return `true`, then `find_or_last()`
+    /// returns [`Some(element)`]. If they all return `false`, it returns the same as if [`last`]
+    /// was invoked instead.
+    ///
+    /// `find_or_last()` is short-circuiting; in other words, it will stop processing as soon as the
+    /// closure returns `true`.
+    ///
+    /// Because `find_or_last()` takes a reference, and many iterators iterate over references, this
+    /// leads to a possibly confusing situation where the argument is a double reference. You can
```

Comment:
```suggestion
    /// leads to a special situation where the argument is a double reference. You can
```
Recommendation to remove passive wording.

Reply:
I copied the wording from `Iterator::find`. Shall I change it there as well? It also present in some other functions, e.g. `Iterator::filter`.

Reply:
It's simply a matter of style. I realize that wording which suggests probable 'confusion' is commonly used throughout Rust code and documentation. 

FWIW I'm not in any position to be making requests of contributors :)